Tests/ConfigurationTests/ConfigurationManagerTests.swift

```diff
@@ -152,37 +151,43 @@ final class ConfigurationManagerTests: XCTestCase {
         let managerA = makeConfigurationManager()
         let managerB = makeConfigurationManager()
 
-        var e: XCTestExpectation? = expectation(description: "ConfigManager B updated")
+        var e = XCTestExpectation(description: "ConfigManager B updated")
         managerB.onDependenciesUpdated = {
-            e?.fulfill()
-            e = nil
+            e.fulfill()
         }
 
         var configData = Data("Privacy Config".utf8)
         MockURLProtocol.requestHandler = { _ in (HTTPURLResponse.ok, configData) }
         await managerA.refreshNow()
-        await fulfillment(of: [e!], timeout: 2)
+        await fulfillment(of: [e], timeout: 2)
 
         XCTAssertNotNil(MockURLProtocol.lastRequest)
         XCTAssertEqual(managerB.dependencyProvider.privacyConfigData, configData)
         XCTAssertEqual(managerB.dependencyProvider.privacyConfigEtag, HTTPURLResponse.testEtag)
 
         MockURLProtocol.lastRequest = nil
-        e = expectation(description: "ConfigManager A updated")
+        e = XCTestExpectation(description: "ConfigManager A updated")
         managerB.onDependenciesUpdated = nil
         managerA.onDependenciesUpdated = {
-            e?.fulfill()
-            e = nil
+            e.fulfill()
         }
 
         configData = Data("Privacy Config 2".utf8)
         MockURLProtocol.requestHandler = { _ in (HTTPURLResponse.ok, configData) }
         await managerB.refreshNow()
-        await fulfillment(of: [e!], timeout: 2)
+        await fulfillment(of: [e], timeout: 2)
 
         XCTAssertNotNil(MockURLProtocol.lastRequest)
-        XCTAssertEqual(managerA.dependencyProvider.privacyConfigData, configData)
-        XCTAssertEqual(managerA.dependencyProvider.privacyConfigEtag, HTTPURLResponse.testEtag)
+
+        let configDataExpectation = XCTNSPredicateExpectation(predicate: NSPredicate { _, _ in
+            return managerA.dependencyProvider.privacyConfigData == configData
+        }, object: .none)
+
+        let configEtagExpectation = XCTNSPredicateExpectation(predicate: NSPredicate { _, _ in
+            return managerA.dependencyProvider.privacyConfigEtag == HTTPURLResponse.testEtag
+        }, object: .none)
```

Comment:
This test has a race condition related to file coordinators that has since become more obvious with Xcode 16. This change to wait for the expected value has fixed it 100% of the time for me, even after hundreds of runs.